### ISSUE TITLE
Getting different feature vector for same image

### ISSUE THREAD

**Issue by mesrh** [kris314/hwnet#3]
Everytime I am getting different feature vector of even same image which sometimes change AP.

Thanks

**Comment by kris314** [comment]
Hi,

Are you using randFlag=True?

Praveen

**Comment by mesrh** [comment]
No, I am using randFlag=False.
Also, I am trying to fine-tune a model but don't know about which loss function should I use(as nothing is specified).

**Comment by kris314** [comment]
Can you share the command that you use running the feature computation, along with the test annotation file?

Regarding fine tuning the model, we haven't provided the training code. The current repository  only provide codes for feature computation. 

Praveen

**Comment by mesrh** [comment]
I used this command for  feature computation:
```
 python3  hwnet-feat.py --annFile ../ann/test_ann.txt --pretrained_file pretrained/iam-model.t7 --img_folder ../wordImages/ --testAug --exp_dir output/ --exp_id iam-test-0 
```

The test annotation file is as follows:
```
m01-049-02-00.png heather 9490 1
m01-049-02-01.png and 38 1
m01-049-02-02.png steve 9522 1
m01-049-02-03.png stood 3739 1
m01-049-02-04.png aghast 9559 1
```


I haven't change anything except I am loading pre-trained model on my CPU device (by adding map_location='cpu' param)

```
checkpoint = torch.load(args.pretrained_file, map_location='cpu')
```

Running  hwnet-feat.py and then eval.py returning me different feature vector for the same image every time (like Stochastic model),  which result in kdt.query returning different distance and index .

**Comment by kris314** [comment]
Thanks for pointing out.  

Yes, the features are coming different in each computation, however this will not affect the quality of features computed. This was due to the augmentation of the rendered image with some amount of translation, which was important while training.

I have now updated the code to switch this off while testing. Please test again.

**Comment by mesrh** [comment]
Thank you for updating the code.
